simconvoi.cc

```diff
@@ -5037,21 +5037,29 @@ const char* convoi_t::send_to_depot_immediately(bool local)
 	route_t *route = new route_t();
 	koord3d home = koord3d::invalid;
 	vehicle_t *v = front();
-	FOR(slist_tpl<depot_t*>, const depot, depot_t::get_depot_list()) {
-		if (depot->get_waytype() != v->get_desc()->get_waytype()  ||  depot->get_owner() != get_owner()) {
-			continue;
-		}
-		koord3d pos = depot->get_pos();
+	koord3d next_pos = schedule->get_current_entry().pos;
+	if (  world()->lookup(next_pos)->get_depot() && v->calc_route(get_pos(), next_pos, 50, route)  ) {
+		// if this convoy is already going to the depot, it will be teleported to that depot.
+		sim::swap(shortest_route, route);
+		home = next_pos;
```

Comment:
ルートを計算する必要はないように見えます。

```suggestion
	const koord3d next_pos = schedule->get_current_entry().pos;
	if (  world()->lookup(next_pos)->get_depot()  ) {
		// if this convoy is already going to the depot, it will be teleported to that depot.
		home = next_pos;
```